alonet/callbacks/object_detector_callback.py

```diff
@@ -142,29 +147,51 @@ def log_masks(self, frames: list, pred_masks: list, trainer: pl.trainer.trainer.
 
             # 2. Send masks to wandb (expend more time process)
             # Retrive label names
-            labels_names = frames.segmentation[b].labels.labels_names
-            if labels_names is not None:
-                labels_names = {i: name for i, name in enumerate(labels_names)}
-
             target_masks = frames.segmentation[b]
+            labels_names = target_masks.labels.labels_names
+            if labels_names is not None:
+                if color_by_cat:
+                    labels_gt = {i: name for i, name in enumerate(labels_names)}
+                    labels_pred = labels_gt
+                else:  # TODO: synchronize colors by matcher (?)
+                    labels = target_masks.labels.cpu().numpy().astype("int")
+                    labels_gt = {
+                        i + VOID_CLASS_ID + 1: labels_names[id]
+                        for i, id in enumerate(labels)
+                        if id < len(labels_names)
+                    }
+                    labels = p_mask.labels.cpu().numpy().astype("int")
+                    labels_pred = {
+                        i + VOID_CLASS_ID + 1: labels_names[id]
+                        for i, id in enumerate(labels)
+                        if id < len(labels_names)
+                    }
 
             frame = frames[b].detach().norm255().cpu().type(torch.uint8).rename(None)
             frame = frame.permute([1, 2, 0]).contiguous().numpy()
 
             # Get panoptic view
-            target_masks = target_masks.mask2id()
-            p_mask = p_mask.mask2id()
-            target_masks[target_masks == -1] = len(labels_names)  # Background N/A
-            p_mask[p_mask == -1] = len(labels_names)  # Background N/A
+            target_masks = target_masks.mask2id(return_cats=color_by_cat)
+            p_mask = p_mask.mask2id(return_cats=color_by_cat)
+            if VOID_CLASS_ID < 0:
```

Comment:
what is VOID_CLASS_ID ?

Reply:
It is a ID-constant  used by mask to set every pixel that have not prediction into a common value. This value is used in compute_pq, callbacks and aloscene.mask. For that, I preferred to save it in a common file, and used it by import